ooniprobe/Engine/PESession.m

```diff
@@ -0,0 +1,39 @@
+#import "PESession.h"
+
+@implementation PESession
+
+- (id)initWithConfig:(OONISessionConfig *)config error:(NSError **)error{
+    self = [super init];
+    if (self) {
+        self.session = OonimkallNewSession([config toOonimkallSessionConfig], error);
+    }
+    return self;
+}
+
+- (OONIGeolocateResults*) geolocate:(OONIContext*) ctx error:(NSError **)error{
+    return [[OONIGeolocateResults alloc] initWithResults:
+            [self.session geolocate:ctx.ctx error:error]];
+
+}
+
+- (void) maybeUpdateResources:(OONIContext*) ctx error:(NSError **)error{
+    [self.session maybeUpdateResources:ctx.ctx
+                                 error:error];
+}
+
+- (OONIContext*) newContext{
+    return [self newContextWithTimeout:-1];
+}
+
+- (OONIContext*) newContextWithTimeout:(long)timeout{
+    return [[OONIContext alloc] initWithContext:
+            [self.session newContextWithTimeout:timeout]];
+
+}
+
+- (OONISubmitResults*) submit:(OONIContext*) ctx measurement:(NSString*) measurement error:(NSError **)error{
+    return [[OONISubmitResults alloc] initWithResults:[self.session submit:ctx.ctx measurement:measurement error:error]];
```

Comment:
```suggestion
    OonimkallSubmitResults *r = self.session submit:ctx.ctx measurement:measurement error:error];
    return (r != nil) ? ([[OONISubmitResults alloc] initWithResults:r]) : nil;
```

Reply:
Why this?

Reply:
We never check if the returned `OonimkallSubmitResults` but we only check the error

Reply:
While that is true, the typical Apple convention is to return either `nil` with an error or a valid value. Because that convention is used by many system APIs as well as APIs maintained by us, not using this convention here is surprising and fragile. For example, if I was to change the code, I would most likely assume that the API only returns `nil` in case of error. This is even more important if we consider that the error argument might in principle be `nil`, leading a developer with no way of assessing whether the result is successful.